cmake/Modules/FindComputeCpp.cmake

```diff
@@ -292,33 +292,36 @@ function(__build_spir targetName sourceFile binaryDir fileCounter)
   # Add a dependency on the integration header
   add_dependencies(${targetName} ${headerTargetName})
 
+  # This property can be set on a per-target basis to indicate that the
+  # integration header should appear after the main source listing
+  get_property(includeAfter TARGET ${targetName}
+      PROPERTY COMPUTECPP_INCLUDE_AFTER)
+
+  if(includeAfter)
+    # Change the source file to the integration header - e.g.
+    # g++ -c source_file_name.cpp.sycl
+    set_property(TARGET ${targetName} PROPERTY SOURCES ${outputSyclFile})
+    # CMake/gcc don't know what language a .sycl file is, so tell them
+    set_property(SOURCE ${outputSyclFile} PROPERTY LANGUAGE CXX)
```

Comment:
you should set LANGUAGE_VERSION here too

Reply:
We don't set the language version in FindComputeCpp.cmake any more, it was occasionally conflicting with what downstream users were using. Instead it is set once globally in the root CMakeLists.txt file. Arguably there should be a better solution than this - should I try to grab the language version of the original source file, then set the integration header to that version?

Reply:
I am concern that Cmake will not set the right language version in this case since it is technically not a C++ file, although you are forcing it to be with the language property.  I tend not to trust CMake implicit decisions in most cases and mark it explicit, maybe by retrieving the property and setting it?

Reply:
OK, I will add this.

Reply:
After discussion, it appears that because the language version is a property of the target, and because this code is similarly modifying the target, this should suffice (and indeed the flag -std=c++14 turns up in the output from CMake).